Add hardening flags to compilation. Fix a format string in order to do
so.
--- a/Makefile
+++ b/Makefile
@@ -37,7 +37,9 @@
 WINCC = $(RAW_CC) $(RAW_FLAGS) -D__WIN32
 
 # Generic "how to compile C files"
-CC = $(RAW_CC) $(RAW_FLAGS) -D__UNIX
+CC = $(RAW_CC) $(RAW_FLAGS) -D__UNIX $(shell dpkg-buildflags --get CFLAGS)\
+	$(shell dpkg-buildflags --get CPPFLAGS) \
+	$(shell dpkg-buildflags --get LDFLAGS)	
 .c.o:   
 	$(CC) -c $<
 
--- a/extract.c
+++ b/extract.c
@@ -2145,7 +2145,7 @@
 			ret_time->tm_sec);
 	chop(ascii_time);
 
-	sprintf(comment, ascii_time);
+	sprintf(comment, "%s", ascii_time);
 	strcat(needle->comment, comment);
 	exe_char = htos(&foundat[22], FOREMOST_LITTLE_ENDIAN);
 	if (exe_char & 0x2000)
